apps/web/src/hooks/use-stored-state.ts

```diff
@@ -0,0 +1,25 @@
+import * as React from "react";
+
+export const useStoredState = <T extends string | undefined>(
+  key: string,
+  defaultValue: T,
+): [state: T, setState: (newValue: T) => void] => {
+  const [state, setState] = React.useState<T>(defaultValue);
+  React.useEffect(() => {
+    const storedValue = localStorage.getItem(key);
+    if (storedValue) {
+      setState(storedValue as T);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
why is that? I believe we should have the `key` in here right? I'm usually against disabling exhaustive deps lint rule unless it's for very specific use cases

Reply:
Agreed, just fixed it. I think I was having a few re-render issues and thought this was the cause of it, but the key should be a dependency there.